carbon-p2-plugin/src/main/java/org/wso2/maven/p2/utils/FileManagementUtil.java

```diff
@@ -0,0 +1,291 @@
+import java.util.zip.ZipInputStream;
+import java.util.zip.ZipOutputStream;
+
+/**
+ * Util class which handle file manipulation operations.
+ */
+public class FileManagementUtil {
+    private static final int BUFFER = 2048;
+
+
+    /**
+     * For a given location and the profile, returns the File object represented by config.ini
+     *
+     * @param destination path pointing the [carbon product]/repository/components folder
+     * @param profile     name of the profile
+     * @return the config.ini file for the Profile
+     */
+    public static File getProfileConfigIniFile(String destination, String profile) {
+        return new File(destination + File.separator + profile + File.separator + "configuration" +
+                File.separator + "config.ini");
+    }
+
+    /**
+     * Updated the given config.ini file with a given key value pair.
+     *
+     * @param configIniFile File object representing the config.ini
+     * @param propKey       property key
+     * @param value         property value
+     */
+    public static void changeConfigIniProperty(File configIniFile, String propKey, String value, Log log) {
+        Properties prop = new Properties();
+
+        try (InputStream inputStream = new FileInputStream(configIniFile)) {
+            prop.load(inputStream);
+            prop.setProperty(propKey, value);
+            try (OutputStream outputStream = new FileOutputStream(configIniFile)) {
+                prop.store(outputStream, null);
+            }
+        } catch (IOException e) {
+            log.warn(e);
+        }
+    }
+
+    /**
+     * Zip a give folder to a give output zip file.
+     *
+     * @param srcFolder   source folder
+     * @param destZipFile path to the output zip file
+     */
+    public static void zipFolder(String srcFolder, String destZipFile, Log log) {
+        try (FileOutputStream fileWriter = new FileOutputStream(destZipFile);
+             ZipOutputStream zip = new ZipOutputStream(fileWriter)) {
+            addFolderContentsToZip(srcFolder, zip, log);
+        } catch (IOException e) {
+            log.warn(e);
+        }
+
+
+    }
+
+    private static void addToZip(String path, String srcFile, ZipOutputStream zip, Log log) {
+        File folder = new File(srcFile);
+        if (folder.isDirectory()) {
+            addFolderToZip(path, srcFile, zip, log);
+        } else {
+            // Transfer bytes from in to out
+            byte[] buf = new byte[1024];
+            int len;
+            try (FileInputStream inputStream = new FileInputStream(srcFile)) {
+
+                if (path.trim().equals("")) {
+                    zip.putNextEntry(new ZipEntry(folder.getName()));
+                } else {
+                    zip.putNextEntry(new ZipEntry(path + "/" + folder.getName()));
+                }
+                while ((len = inputStream.read(buf)) > 0) {
+                    zip.write(buf, 0, len);
+                }
+                inputStream.close();
```

Comment:
Not necessary to close the stream as the stream is managed and closed by try-with-resources. This the line inputStream.close(); is removed in https://github.com/wso2/carbon-maven-plugins/pull/25